src/main/java/picard/sam/BamErrorMetric/CollectBamErrorMetrics.java

```diff
@@ -0,0 +1,470 @@
+                "confidence interval where the user is relatively certain that the polymorphic sites are known and accurate. " +
+                "These two inputs are provided as a VCF and INTERVALS. The program will only process sites that are in the " +
+                "intersection of the interval lists in the INTERVALS argument as long as they are not polymorphic in the " +
+                "VCF.\n\n",
+        oneLineSummary = "Program to collect error metrics on bases stratified in various ways.",
+        programGroup = DiagnosticsAndQCProgramGroup.class
+)
+public class CollectBamErrorMetrics extends CommandLineProgram {
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "Input SAM or BAM file.")
+    public File INPUT;
+
+    @Argument(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "Base name for output files. Actual file names will be " +
+            "generated from the basename and suffixes from the ERROR and STRATIFIER by adding a '.' and then " +
+            "error_by_stratifier[_and_stratifier]* where 'error' is ERROR's extension, and 'stratifier' is STRATIFIER's suffix. " +
+            "For example, an ERROR_METRIC of ERROR:BASE_QUALITY:GC_CONTENT will produce an extension '.error_by_base_quality_and_gc'. " +
+            "The suffixes can be found in the documentation for ERROR_VALUE and SUFFIX_VALUE." )
+    public File OUTPUT;
+
+    @Argument(doc = "Errors to collect in the form of \"ERROR(:STRATIFIER)*\". " +
+            "To see the values available for ERROR and STRATIFIER look at the documentation for the arguments " +
+            "ERROR_VALUE and STRATIFIER_VALUE.")
+    public List<String> ERROR_METRICS = new ArrayList<>(CollectionUtil.makeList(
+            "ERROR",
+            "ERROR:BASE_QUALITY",
+            "ERROR:INSERT_LENGTH",
+            "ERROR:GC_CONTENT",
+            "ERROR:READ_DIRECTION",
+            "ERROR:PAIR_ORIENTATION",
+            "ERROR:HOMOPOLYMER",
+            "ERROR:BINNED_HOMOPOLYMER",
+            "ERROR:CYCLE",
+            "ERROR:READ_ORDINALITY",
+            "ERROR:READ_ORDINALITY:CYCLE",
+            "ERROR:READ_ORDINALITY:HOMOPOLYMER",
+            "ERROR:READ_ORDINALITY:GC_CONTENT",
+            "ERROR:READ_ORDINALITY:PRE_DINUC",
+            "ERROR:MAPPING_QUALITY",
+            "ERROR:READ_GROUP",
+            "ERROR:MISMATCHES_IN_READ",
+            "ERROR:ONE_BASE_PADDED_CONTEXT",
+            "OVERLAPPING_ERROR",
+            "OVERLAPPING_ERROR:BASE_QUALITY",
+            "OVERLAPPING_ERROR:INSERT_LENGTH",
+            "OVERLAPPING_ERROR:READ_ORDINALITY",
+            "OVERLAPPING_ERROR:READ_ORDINALITY:CYCLE",
+            "OVERLAPPING_ERROR:READ_ORDINALITY:HOMOPOLYMER",
+            "OVERLAPPING_ERROR:READ_ORDINALITY:GC_CONTENT"));
+
+    @Argument(doc = "A fake argument used to show what the options of ERROR (in ERROR_METRICS) are.", optional = true)
+    public BaseErrorCalculation.Errors ERRORS_VALUE = null;
```

Comment:
`= null` is unnecessary in Java as that is the default value of an object. same is true of `= 0` below for `long`